src/xcomponentAPI.js

```diff
@@ -1,26 +1,26 @@
 
-define(["communication/connection", "parser", "configuration/xcomponentConfiguration", "communication/xcomponentPublisher", "rx"], function (Connection, Parser, XComponentConfiguration, Publisher, Rx) {
+define(["communication/connection", "parser", "configuration/xcomponentConfiguration", "rx"], function (Connection, Parser, XComponentConfiguration, Rx) {
     "use strict";
 
-    var Factory = function () {
-        this.createXComponentApi = function (serverUrl, callback, callbackError) {
-            var xcApiString = XComponentConfiguration.getXcApi();
-            var parser = new Parser(xcApiString);
-            var publisher = new Publisher(parser);
-            var connection = new Connection(serverUrl, callback, callbackError, publisher);
-            var api = new XComponentAPI(connection);
-            return api;
-        }
+
+    var createApi = function (serverUrl, callback) {
+        var xcApiString = XComponentConfiguration.getXcApi();
```

Comment:
weird
We can try something like this:
var configuration = new xcConfiguration(parser)
configuration.init()